### PR TITLE
40_refactor_camera_class

### PR DESCRIPTION
**Pull request by wkmeijer**
Split camera2basic fragment into three classes.

### CODE REVIEW

File: app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java [hunk 1]
```diff
@@ -68,68 +55,55 @@
...
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.Comparator;
 import java.util.List;
 import java.util.concurrent.Semaphore;
-import java.util.concurrent.TimeUnit;
 
-public class Camera2BasicFragment extends Fragment
-        implements FragmentCompat.OnRequestPermissionsResultCallback {
+public class Camera2BasicFragment extends Fragment implements FragmentCompat.OnRequestPermissionsResultCallback {
```
Comment (jonathanraes) [inline app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java]: maybe a rename of Camera2BasicFragment class is a good idea

File: app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java [hunk 2]
```diff
@@ -68,68 +55,55 @@
...
-    private static final String FRAGMENT_DIALOG = "dialog";
-
-    boolean mIsStateAlreadySaved = false;
-    boolean mPendingShowDialog = false;
-
     static {
         ORIENTATIONS.append(Surface.ROTATION_0, 90);
         ORIENTATIONS.append(Surface.ROTATION_90, 0);
         ORIENTATIONS.append(Surface.ROTATION_180, 270);
         ORIENTATIONS.append(Surface.ROTATION_270, 180);
```
Comment (jonathanraes) [inline app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java]: I believe this is unused

File: app/src/main/java/com/digitalvotingpass/camera/CameraHandler.java [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        return mPreviewSize;
+    }
+
+    /**
+     * Find out if we need to swap dimensions to get the preview size relative to sensor coordinate.
+     * @param activity - The associated activity from which the camera is loaded.
+     * @param characteristics - CameraCharacteristics corresponding to the current started cameradevice
+     * @return swappedDimensions - A boolean value indicating if the the dimensions need to be swapped.
+     */
+    private boolean needSwappedDimensions(CameraCharacteristics characteristics) {
```
Comment (jonathanraes) [inline app/src/main/java/com/digitalvotingpass/camera/CameraHandler.java]: The activity param doesnt exist anymore. Also this method can probably also be moved to CameraSupport

**Review by jonathanraes — CHANGES_REQUESTED** [review]
Greatly reduces Camera2BasicFragment in sie which is awesome.
I think however in order to please SIG we can still break some of the camera related methods into multiple shorter ones.
Also kind of unsure about CameraSupport, since all its methods are only used once and it doenst seem to have a clear single responsibility

File: app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java (hunk 2 above)
Comment (wkmeijer) [reply app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java]: You are correct

File: app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java (hunk 1 above)
Comment (wkmeijer) [reply app/src/main/java/com/digitalvotingpass/camera/Camera2BasicFragment.java]: Renamed to CameraFragment

File: app/src/main/java/com/digitalvotingpass/camera/CameraFragment.java [hunk 4]
```diff
@@ -370,7 +370,7 @@ public synchronized void scanResultFound(final Mrz mrz) {
             }
             Intent returnIntent = new Intent();
             DocumentData data = mrz.getPrettyData();
-            returnIntent.putExtra("result", data);
+            returnIntent.putExtra("docData", data);
```
Comment (klikooo) [inline app/src/main/java/com/digitalvotingpass/camera/CameraFragment.java]: we have a final static string for this: DocumentData.identifier

Comment (wkmeijer) [reply app/src/main/java/com/digitalvotingpass/camera/CameraFragment.java]: fixed

**Review by jonathanraes — APPROVED** [review]

**Pull request merged**